standard/date_ext.lua

```diff
@@ -52,7 +52,9 @@ function DateExt.readTimestamp(dateInput)
 
 	-- Extracts the '-4:00' out of <abbr data-tz="-4:00" title="Eastern Daylight Time (UTC-4)">EDT</abbr>
 	local tzTemplateOffset = dateInput:match('data%-tz%=[\"\']([%d%-%+%:]+)[\"\']')
-	local datePart = (mw.text.split(dateInput, '<', true)[1]):gsub('-', '')
+	local datePart = (mw.text.split(dateInput, '<', true)[1])
+		:gsub('-', '')
+		:gsub('T', '')
```

Comment:
I'd still suggest something like this instead, which only trims the dashes, but does not entirely remove them
```suggestion
	local datePart = mw.text.trim(mw.text.split(dateInput, '<', true)[1], '-\t\r\n\f\v ')
```

Reply:
doesn't work
trim only gets rid of stuff at the start and end of the string
the dash that causes issues is between the date and the time so the trim doesn't do anything to it

Reply:
we could do `:gsub('%s*%-%s*(%d+):', 'T%1:')` to replace the dash before the time with a T

Reply:
Ah right, i thought it only separates the Abbr/TZ stuff, but it's also before the time itself.
Let's go with the current state of PR then, better to maintain than lua patterns.